apollo-federation/src/sources/connect/json_selection/methods.rs

```diff
@@ -0,0 +1,2151 @@
+use apollo_compiler::collections::IndexMap;
+use apollo_compiler::collections::IndexSet;
+use lazy_static::lazy_static;
+use serde_json_bytes::serde_json::Number;
+use serde_json_bytes::ByteString;
+use serde_json_bytes::Map as JSONMap;
+use serde_json_bytes::Value as JSON;
+
+use super::helpers::json_type_name;
+use super::immutable::InputPath;
+use super::lit_expr::LitExpr;
+use super::ApplyTo;
+use super::ApplyToError;
+use super::MethodArgs;
+use super::PathList;
+use super::VarsWithPathsMap;
+
+type ArrowMethod = fn(
+    // Method name
+    &str,
+    // Arguments passed to this method
+    &Option<MethodArgs>,
+    // The JSON input value (data)
+    &JSON,
+    // The variables
+    &VarsWithPathsMap,
+    // The input_path (may contain integers)
+    &InputPath<JSON>,
+    // The rest of the PathList
+    &PathList,
+    // Errors
+    &mut IndexSet<ApplyToError>,
```

Comment:
You can add names to these.

```suggestion
    // Method name
    name: &str,
    // Arguments passed to this method
    args: &Option<MethodArgs>,
    // The JSON input value (data)
    data: &JSON,
    // The variables
    vars: &VarsWithPathsMap,
    // The input_path (may contain integers)
    path: &InputPath<JSON>,
    // The rest of the PathList
    rest: &PathList,
    // Errors
    errors: &mut IndexSet<ApplyToError>,
```

Reply:
https://github.com/apollographql/router/pull/5762/commits/94a0186b1cb4aa97ca5c39123bf21a7f9711feb3